### PR TITLE
Feature/lp 9 email validation

### PR DESCRIPTION
**Pull request by NomasChuy**
Hi team this are my tasks, I decided to make them all in a single branch, sorry for the inconvenience, but I think it was better if I did them all at once because it was all related to each task.

https://devaptivist.atlassian.net/jira/software/projects/LP/boards/4?selectedIssue=LP-96
https://devaptivist.atlassian.net/jira/software/projects/LP/boards/4?selectedIssue=LP-97
https://devaptivist.atlassian.net/jira/software/projects/LP/boards/4?selectedIssue=LP-98

In summary, here are the bullet points that I needed to cover:

1. The email text field should have an appropriate placeholder text to guide the user in filling out the field.
2. The email text field should have validation to ensure that the user has entered a valid email address format, including an "@" symbol and a domain name.
3. The email text field should pass directly to the password field when the enter button is clicked
4. The email and password text field should be able to handle and display any error messages or validation errors in a clear and understandable way.
5. The password field should be visible to the user

Here is a preview of each point:

1.-
image

2.-
image

3.-
image
image

4.-
image

5.-
image

### CODE REVIEW

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt [hunk 1]
```diff
@@ -222,7 +231,7 @@ fun LpOutlinedTextFieldInput(
             unfocusedBorderColor = MaterialTheme.colorScheme.onPrimary,
             errorBorderColor = MaterialTheme.colorScheme.error,
             errorLabelColor = MaterialTheme.colorScheme.error,
-            errorSupportingTextColor = MaterialTheme.colorScheme.error
+            errorSupportingTextColor = MaterialTheme.colorScheme.error,
```
Comment (carlosph816) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt]: we can delete the "," in the end

File: feature/login/src/main/java/com/iteneum/login/presentation/LoginViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
+package com.iteneum.login.presentation
+
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import javax.inject.Inject
+
```
Comment (carlosph816) [inline feature/login/src/main/java/com/iteneum/login/presentation/LoginViewModel.kt:9]: You are missing the information of this class and author

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt [hunk 3]
```diff
@@ -24,7 +29,7 @@ import com.iteneum.designsystem.components.phonenumbertext.PhoneNumberTransforma
  * This function creates a password OutlinedTextField
  * @param modifier Set component modifier
  * @param value Current password value
- * @param isValid Validate if text is valid
+ * @param isPasswordError Validate if text is valid
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt]: please update the description of the param

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt [hunk 4]
```diff
@@ -137,7 +145,7 @@ fun LpOutlinedTextField(
  * This function creates an Email OutlinedTextField
  * @param modifier Set component modifier
  * @param label Text label
- * @param isValid Validate if text is valid
+ * @param isEmailError Validate if text is valid
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt]: Please add the new param to the documentation

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt [hunk 5]
```diff
@@ -149,8 +157,10 @@ fun LpOutlinedTextField(
 fun LpOutlinedTextFieldMail(
     modifier: Modifier,
     label: String,
-    isValid: Boolean,
+    placeholder: String,
+    isEmailError: Boolean,
     supportTextError: String,
+    onImeActionPerformed: () -> Unit,
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:199]: please add the `onImeActionPerformed ` to the documentation

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt [hunk 6]
```diff
@@ -163,27 +173,26 @@ fun LpOutlinedTextFieldMail(
         },
         modifier = modifier,
         label = { Text(label, style = MaterialTheme.typography.bodyMedium) },
+        placeholder = { Text(placeholder) },
```
Comment (melissaquezada94) [inline core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:214]: please add a gray color to the label and placeholder

File: feature/login/src/main/java/com/iteneum/login/presentation/LoginView.kt [hunk 7]
```diff
@@ -68,41 +76,46 @@ fun LoginView() {
                     end.linkTo(parent.end)
                 },
             label = stringResource(R.string.lv_email),
-            isValid = false,
-            supportTextError = stringResource(R.string.lv_support_text_error),
-            onValueChange = { /*TODO("This will change the value of the email")*/ }
+            placeholder = stringResource(R.string.lv_example_email),
```
Comment (melissaquezada94) [inline feature/login/src/main/java/com/iteneum/login/presentation/LoginView.kt:95]: please add a gray color to the placeholder

**Review by carlosph816 — APPROVED** [review]
Well done!

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt (hunk 6 above)
Comment (NomasChuy) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:214]: Hi Meli!, I think I did defined it as gray, you can check it on line 185 if I'm not mistaken, and as for the label the color was already defined on line 181, let me know if I'm mistaken :smiley:

File: feature/login/src/main/java/com/iteneum/login/presentation/LoginView.kt (hunk 7 above)
Comment (NomasChuy) [reply feature/login/src/main/java/com/iteneum/login/presentation/LoginView.kt:95]: Hi Meli! , the color of the placeholder is already gray in the component file, should I add it here as well?

**Review by melissaquezada94 — APPROVED** [review]
well done!

File: core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt (hunk 6 above)
Comment (melissaquezada94) [reply core/designsystem/src/main/java/com/iteneum/designsystem/components/TextField.kt:214]: Amm nop, you missed the color in the text on line 199, we should use the color tertiary on that as well.

File: feature/login/src/main/java/com/iteneum/login/presentation/LoginView.kt [hunk 8]
```diff
@@ -45,15 +42,18 @@ import com.iteneum.login.R
...
+fun LoginView(loginViewModel: LoginViewModel = hiltViewModel()) {
     val sizes = LeasePertTheme.sizes
 
     ConstraintLayout(
         modifier = Modifier
             .fillMaxSize()
 
     ) {
-        val (logo, mail, password, tonalButton, loginWith, socialNetworkLogins, registerButton) = createRefs()
+        val (logo, emailField, passwordField, loginButton, loginWith, socialNetworkLogins, registerButton) = createRefs()
```
Comment (melissaquezada94) [inline feature/login/src/main/java/com/iteneum/login/presentation/LoginView.kt:67]: You should use remember variables to set it to the viewModel

File: feature/login/src/main/java/com/iteneum/login/presentation/LoginViewModel.kt [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+ * the user to a registering screen.
+ *
+ * @author Jesus Lopez
+ */
+@HiltViewModel
+class LoginViewModel @Inject constructor() : ViewModel() {
+    private val _email = MutableStateFlow("")
+    val email: StateFlow<String> = _email
+
+    private val _isEmailError = MutableStateFlow(false)
```
Comment (melissaquezada94) [inline feature/login/src/main/java/com/iteneum/login/presentation/LoginViewModel.kt]: You can define a data class Login with email and password

File: feature/login/src/main/java/com/iteneum/login/presentation/LoginView.kt (hunk 8 above)
Comment (NomasChuy) [reply feature/login/src/main/java/com/iteneum/login/presentation/LoginView.kt:67]: Those variables are needed to create the references for the constraintLayout for the composable view, should I still proceed to change them to remember variables?

**Review by irv205 — APPROVED** [review]

**Pull request merged**